sshd-core/src/main/java/org/apache/sshd/common/io/nio2/Nio2Session.java

```diff
@@ -312,7 +312,7 @@ protected void handleReadCycleCompletion(
                 if (log.isDebugEnabled()) {
                     log.debug("handleReadCycleCompletion({}) Socket has been disconnected (result={}), closing IoSession now", this, result);
                 }
-                close(true);
+                close(false);
```

Comment:
Why `close(false)` ? Let's add some comment explaining so in the future it is not changed back inadvertently